### PR TITLE
Smt12 mariadb forward registration

### PR DESCRIPTION
**Pull request by mcalmer**
Forward registrations to SCC

### CODE REVIEW

**Comment by mcalmer** [comment]
@aaannz: please have a look. If you have a test setup for SMT12, would be good if you could give this a try.

File: script/smt-register [hunk 1]
```diff
@@ -136,17 +137,24 @@ eval
...
+        my $sccreg = SMT::SCCSync->new(vblevel   => $vblevel,
+                                       useragent => $useragent,
+                                       log       => $LOG,
+                                       dbh       => $dbh,
+                                       nccemail  => $nccEmail);
+        my $err = $sccreg->register_systems($sleeptime);
+        if($err != 0)
+        {
+            $exitcode = 1;
+        }
```
Comment (aaannz) [inline script/smt-register]: isn't there missing }?

Comment (mcalmer) [inline script/smt-register]: Correct, fixed with commit 9bf1754d

File: script/smt-report [hunk 2]
```diff
@@ -495,64 +432,22 @@ $LOG = SMT::Utils::openLog($logfile);
 # calculate best report
 my $cfg = SMT::Utils::getSMTConfig();
 
-if ( $cfg->val('NU', 'ApiType', 'NCC') eq 'NCC')
+my $api = $cfg->val('NU', 'ApiType', 'SCC');
```
Comment (aaannz) [inline script/smt-report]: SCC as default?

Comment (mcalmer) [inline script/smt-report]: We can think about it. SCC is the only supported ApiType for SMT12. So why not?
But we may want to change it globally?
Or does it make trouble with the migration?

Comment (aaannz) [inline script/smt-report]: I don't think there is a problem with migration, but we use NCC as default everywhere else so this surprised me.

Comment (mcalmer) [inline script/smt-report]: Ok, I changed it to NCC to keep it consistent everywhere.

**Comment by aaannz** [comment]
LGTM, but I have not tested that yet, working on it right now. Btw can you take a look at #31 ?

**Comment by mcalmer** [comment]
Please tell me if you are finished with the real tests. Or should I merge now ?

File: script/smt-report [hunk 3]
```diff
@@ -495,64 +432,22 @@ $LOG = SMT::Utils::openLog($logfile);
...
+if ( $api eq 'NCC')
 {
-    printLog($LOG, $vblevel, LOG_ERROR, __("NCC is not supported in this version of SMT"));
+    printLog($LOG, $vblevel, LOG_ERROR, __("NCC is not supported by this version of SMT"));
     exit 1;
 }
-
-# sync current data from SCC
-if(!$nosccsync)
+if ( $api eq 'SCC' && $nccreport)
```
Comment (aaannz) [inline script/smt-report]: Compilation fail on undeclared $nccreport.
I noticed --scc option was removed from GetOptions and $sccreport went away with that. Should I removed this whole section or put back --scc only to be able to print this warning? Usage string needs updating too.

Comment (mcalmer) [inline script/smt-report]: remove the whole section. It does not make sense to add an option which only return an error if you use it.

**Pull request merged**